### PR TITLE
lightning: extract loader config for reuse

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #52142

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by lance6716 — APPROVED** [review]

File: br/pkg/lightning/config/config_test.go [hunk 1]
```diff
@@ -1330,6 +1330,8 @@ func TestAdjustBlockSize(t *testing.T) {
 	cfg.TikvImporter.BlockSize = 0
 
 	err := cfg.Adjust(context.Background())
+	// should return "dial tcp 127.0.0.1:10080: connect: connection refused" error,
```
Comment (lyzx2001) [inline br/pkg/lightning/config/config_test.go:1333]: Sorry I think I did not write this test correctly before, the `dial tcp 127.0.0.1:10080: connect: connection refused` error has been tested by `TestAdjustConnectRefused`.

Maybe we can add
```
ts, host, port := startMockServer(t, http.StatusOK,
	`{"port":6666,"advertise-address":"121.212.121.212:5555","path":"34.34.34.34:3434"}`,
)
defer ts.Close()
```
```
cfg.TiDB.Host = host
cfg.TiDB.StatusPort = port
``` 
in this test, and change to `require.NoError(t, err)`?

Comment (D3Hunter) [reply br/pkg/lightning/config/config_test.go:1333]: we can do it in later pr, ok for now

**Review by lyzx2001 — APPROVED** [review]

**Pull request merged**